In [39]:
from pandas import DataFrame, read_csv
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [40]:
Location = r'C:\Users\jwhit\Documents\School\IS362\movies.csv'
movies = pd.read_csv(Location)
movies

Names  Black Panther  A Wrinkle in Time  IT  Red Sparrow  Jumanji
0  Summer              5                  3   5            2        4
1   Tasha              5                  3   4            4        2
2    Jace              5                  4   5            3        3
3    Paul              5                  2   3            5        3
4  Jabari              4                  3   4            5        3

In [41]:
movies=movies.set_index('Names')
movies

Black Panther  A Wrinkle in Time  IT  Red Sparrow  Jumanji
Names                                                             
Summer              5                  3   5            2        4
Tasha               5                  3   4            4        2
Jace                5                  4   5            3        3
Paul                5                  2   3            5        3
Jabari              4                  3   4            5        3

### Above we change the index of the dataframe to the Names column

In [42]:
movies.loc["Average"]=movies.mean(axis=0)
movies

Black Panther  A Wrinkle in Time   IT  Red Sparrow  Jumanji
Names                                                               
Summer             5.0                3.0  5.0          2.0      4.0
Tasha              5.0                3.0  4.0          4.0      2.0
Jace               5.0                4.0  5.0          3.0      3.0
Paul               5.0                2.0  3.0          5.0      3.0
Jabari             4.0                3.0  4.0          5.0      3.0
Average            4.8                3.0  4.2          3.8      3.0

### Above we added a row with the average rating for each movie to the data frame

In [43]:
movies["Participant Average"]=movies.mean(axis=1)
movies

Black Panther  A Wrinkle in Time   IT  Red Sparrow  Jumanji  \
Names                                                                  
Summer             5.0                3.0  5.0          2.0      4.0   
Tasha              5.0                3.0  4.0          4.0      2.0   
Jace               5.0                4.0  5.0          3.0      3.0   
Paul               5.0                2.0  3.0          5.0      3.0   
Jabari             4.0                3.0  4.0          5.0      3.0   
Average            4.8                3.0  4.2          3.8      3.0   

         Participant Average  
Names                         
Summer                  3.80  
Tasha                   3.60  
Jace                    4.00  
Paul                    3.60  
Jabari                  3.80  
Average                 3.76

### Here we added a column with the average of each participant 

In [44]:
movies_norm=(movies-movies.min())/(movies.max()-movies.min())
movies_norm

Black Panther  A Wrinkle in Time   IT  Red Sparrow  Jumanji  \
Names                                                                  
Summer             1.0                0.5  1.0     0.000000      1.0   
Tasha              1.0                0.5  0.5     0.666667      0.0   
Jace               1.0                1.0  1.0     0.333333      0.5   
Paul               1.0                0.0  0.0     1.000000      0.5   
Jabari             0.0                0.5  0.5     1.000000      0.5   
Average            0.8                0.5  0.6     0.600000      0.5   

         Participant Average  
Names                         
Summer                   0.5  
Tasha                    0.0  
Jace                     1.0  
Paul                     0.0  
Jabari                   0.5  
Average                  0.4

### The data has been normalized with a max value of 1 and a minimum value of 0, when viewed this way the data can become confusing because some people may not interpert the value of 0.3333 or 0.66667

In [46]:
movies

Black Panther  A Wrinkle in Time   IT  Red Sparrow  Jumanji  \
Names                                                                  
Summer             5.0                3.0  5.0          2.0      4.0   
Tasha              5.0                3.0  4.0          4.0      2.0   
Jace               5.0                4.0  5.0          3.0      3.0   
Paul               5.0                2.0  3.0          5.0      3.0   
Jabari             4.0                3.0  4.0          5.0      3.0   
Average            4.8                3.0  4.2          3.8      3.0   

         Participant Average  
Names                         
Summer                  3.80  
Tasha                   3.60  
Jace                    4.00  
Paul                    3.60  
Jabari                  3.80  
Average                 3.76

### Above we recall the movies dataframe

In [48]:
movies_std = (movies-movies.mean())/movies.std(ddof=0)
movies_std

Black Panther  A Wrinkle in Time       IT   Red Sparrow   Jumanji  \
Names                                                                        
Summer        0.547723           0.000000  1.17108 -1.690806e+00  1.732051   
Tasha         0.547723           0.000000 -0.29277  1.878673e-01 -1.732051   
Jace          0.547723           1.732051  1.17108 -7.514691e-01  0.000000   
Paul          0.547723          -1.732051 -1.75662  1.127204e+00  0.000000   
Jabari       -2.190890           0.000000 -0.29277  1.127204e+00  0.000000   
Average       0.000000           0.000000  0.00000 -4.171492e-16  0.000000   

         Participant Average  
Names                         
Summer               0.29277  
Tasha               -1.17108  
Jace                 1.75662  
Paul                -1.17108  
Jabari               0.29277  
Average              0.00000

### The dataframe above has been standardized, this topic is confusing to me and I am not really sure how to interpert the results